crates/ra_hir_expand/src/builtin_macro.rs

```diff
@@ -9,26 +9,52 @@ use crate::{
 use crate::quote;
 
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
-pub enum BuiltinExpander {
+pub enum BuiltinFnLikeExpander {
```

Comment:
:+1: 

Can this be generated by `register_builtin`? If we did that we could use `kind as u16` to index into `BUILTIN_MACROS` (with a `[debug_]assert` that `result.kind==kind`). Ideally we would use an array (using `BuiltinFnLikeExpander::$last as usize` for the array size)

I suppose it probably doesn't matter - it depends on how the manual iteration is optimised. I just like looking for some stupid optimisation opportunities.

Reply:
Good idea, I just use macro for all the setup and eliminate the immediate struct too.